Doc/library/contextvars.rst

```diff
@@ -144,21 +144,79 @@ Manual Context Management
    To get a copy of the current context use the
    :func:`~contextvars.copy_context` function.
 
-   Every thread will have a different top-level :class:`~contextvars.Context`
-   object. This means that a :class:`ContextVar` object behaves in a similar
-   fashion to :func:`threading.local` when values are assigned in different
-   threads.
+   Each thread has its own effective stack of :class:`!Context` objects.  The
+   *current context* is the :class:`!Context` object at the top of the current
+   thread's stack.  All :class:`!Context` objects in the stacks are considered
+   to be *entered*.
+
+   *Entering* a context, either by calling the :meth:`~Context.run` method or
+   using the context as a :term:`context manager`, pushes the context onto the
+   top of the current thread's stack, making it the current context.
+
+   *Exiting* from the current context, either by returning from the callback
+   passed to :meth:`~Context.run` or by exiting the :keyword:`with` statement
+   suite, pops the context off of the top of the stack, restoring the current
+   context to what it was before.
+
+   Since each thread has its own context stack, :class:`ContextVar` objects
+   behave in a similar fashion to :func:`threading.local` when values are
+   assigned in different threads.
+
+   Attempting to do either of the following raises a :exc:`RuntimeError`:
+
+   * Entering an already entered context, including contexts entered in
+     other threads.
+   * Exiting from a context that is not the current context.
+
+   After exiting a context, it can later be re-entered (from any thread).
+
+   Any changes to :class:`ContextVar` values via the :meth:`ContextVar.set`
+   method are recorded in the current context.  The :meth:`ContextVar.get`
+   method returns the value associated with the current context.  Exiting a
+   context effectively reverts any changes made to context variables while the
+   context was entered (if needed, the values can be restored by re-entering the
+   context).
 
    Context implements the :class:`collections.abc.Mapping` interface.
 
+   .. versionadded:: 3.14
+      Added support for the :term:`!context management protocol` (:pep:`343`).
+      The value bound to the identifier given in the :keyword:`with` statement's
+      :keyword:`!as` clause (if present) is the :class:`!Context` object itself.
+
+      Example:
```

Comment:
The example should be out of the `.. versionchanged` section, ideally before (we should use `versionchanged` instead of `versionadded` since the latter is only when the object is being added, not when features are added).